releasenotes/notes/improve-svc-start-stop-when-pending-759b6a6285aa9598.yaml

```diff
@@ -0,0 +1,15 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Improve utility functions which start or stop a service to better
+    handle SERVICE_START_PENDING and SERVICE_STOP_PENDING states by simply
+    waiting for the service to complete transition. It will help better
+    handle Windows Agent concurrent execution of`agent.exe start` and
+    `agent.exe restart` commands.
```

Comment:
```suggestion
    Improve utility functions that start or stop a service to better manage the SERVICE_START_PENDING and SERVICE_STOP_PENDING states by waiting for the service transition to complete. This will improve handling of concurrent execution of `agent.exe start` and `agent.exe restart` commands for the Windows Agent.
```